### PR TITLE
Feature/check for updates

### PR DESCRIPTION
**Pull request by itaymmguardicore**
# Feature / Fixes
> Add check for update feature for monkey island

* [X] Have you added an explanation of what your changes do and why you'd like to include them?
* [X] Have you successfully tested your changes locally?

### CODE REVIEW

File: monkey/monkey_island/cc/server_config.json [hunk 1]
```diff
@@ -1,3 +1,5 @@
 {
-  "server_config": "standard"
+  "server_config": "standard",
+  "deployment": "develop",
+  "monkey_version": "1.6.2"
```
Comment (acepace) [inline monkey/monkey_island/cc/server_config.json]: Add newline at end of file?

File: monkey/monkey_island/cc/services/version_update.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        url = VersionUpdateService.VERSION_SERVER_CHECK_NEW_URL % (env.get_deployment(), env.get_version())
+
+        reply = requests.get(url, timeout=15)
+
+        res = reply.json().get('newer_version', None)
+
+        if res is False:
+            return res
+
+        [int(x) for x in res.split('.')]  # raises value error if version is invalid format
```
Comment (acepace) [inline monkey/monkey_island/cc/services/version_update.py]: We can't return new dev version this way? we're forced to return int.int.int.int...

File: monkey/monkey_island/cc/resources/version_update.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+logger = logging.getLogger(__name__)
+
+
+class VersionUpdate(flask_restful.Resource):
+    def __init__(self):
+        super(VersionUpdate, self).__init__()
+
+    @jwt_required()
+    def get(self):
```
Comment (danielguardicore) [inline monkey/monkey_island/cc/resources/version_update.py]: Not sure this should be `jwt_required`

File: monkey/monkey_island/cc/services/version_update.py (hunk 2 above)
Comment (itaymmguardicore) [reply monkey/monkey_island/cc/services/version_update.py]: No such thing as "new dev version"

File: monkey/monkey_island/cc/resources/version_update.py (hunk 3 above)
Comment (itaymmguardicore) [reply monkey/monkey_island/cc/resources/version_update.py]: Fixed

File: monkey/monkey_island/cc/server_config.json (hunk 1 above)
Comment (itaymmguardicore) [reply monkey/monkey_island/cc/server_config.json]: Fixed

File: monkey/monkey_island/cc/environment/__init__.py [hunk 4]
```diff
@@ -13,6 +13,7 @@ class Environment(object):
     _MONGO_URL = os.environ.get("MONKEY_MONGO_URL", "mongodb://localhost:27017/monkeyisland")
     _DEBUG_SERVER = False
     _AUTH_EXPIRATION_TIME = timedelta(hours=1)
+    _MONKEY_VERSION = "1.6.2"
```
Comment (danielguardicore) [inline monkey/monkey_island/cc/environment/__init__.py]: Why is it hardcoded here?

**Pull request merged**